activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,10 @@
+*   Warn about changing `query_constraints:` behavior
+
+    `query_constraints:` association will change behavior in the future versions of Rails
+    and applications are advised to switch to `foreign_key:` to preserve the current behavior.
```

Comment:
I think it's confusing to say that behavior will change but not to what. Applications just need to know to switch, not that there's some other future change.

Reply:
Yeah I agree, since we can't promise anything concrete it's best to just ask applications to migrate to avoid any confusion in the future. I'll drop the "future versions" bit